common/models/role.js

```diff
@@ -187,38 +187,78 @@ module.exports = function(Role) {
       return;
     }
 
+    var isMultiOwnerAllowed = modelClass.settings.ownerRelations || false;
     modelClass.findById(modelId, function(err, inst) {
       if (err || !inst) {
         debug('Model not found for id %j', modelId);
         if (callback) callback(err, false);
         return;
       }
-      debug('Model found: %j', inst);
-      var ownerId = inst.userId || inst.owner;
-      // Ensure ownerId exists and is not a function/relation
-      if (ownerId && 'function' !== typeof ownerId) {
-        if (callback) callback(null, matches(ownerId, userId));
-        return;
+
+      if (!isMultiOwnerAllowed) {
+        // Here is the old logic owner resolving
+        debug('Model found: %j', inst);
+        var ownerId = inst.userId || inst.owner;
+        // Ensure ownerId exists and is not a function/relation
+        if (ownerId && 'function' !== typeof ownerId) {
+          if (callback) callback(null, matches(ownerId, userId));
+          return;
+        } else {
+          // Try to follow belongsTo
+          for (var r in modelClass.relations) {
+            var rel = modelClass.relations[r];
+            if (rel.type === 'belongsTo' && isUserClass(rel.modelTo)) {
+              debug('Checking relation %s to %s: %j', r, rel.modelTo.modelName, rel);
+              inst[r](processRelatedUser);
+              return;
+            }
+          }
+          debug('No matching belongsTo relation found for model %j and user: %j', modelId, userId);
+          if (callback) callback(null, false);
+        }
+
+        function processRelatedUser(err, user) {
+          if (!err && user) {
+            debug('User found: %j', user.id);
+            if (callback) callback(null, matches(user.id, userId));
+          } else {
+            if (callback) callback(err, false);
+          }
+        }
       } else {
-        // Try to follow belongsTo
+        /**
+         * Here is the new way to resolve owner relations
+         * First, fetches `belongsTo` relation (User)
+         * Then, fallback looking for userId / owner
+         */
```

Comment:
If I understand the code correctly, then you are proposing to ignore hard-coded property names like `userId` and `ownerId` when `ownerRelations` is configured. I guess it makes sense, we must describe this difference in our documentation.

Please add unit-test(s) to verify that when `ownerRelations` is set, and there is no relation associated with `userId`/`ownerId`, then these two properties are ignored.